sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpClient.cs

```diff
@@ -189,7 +194,7 @@ internal class AmqpClient : TransportClient
                     clientOptions.CertificateValidationCallback);
 
                 ManagementLink = new FaultTolerantAmqpObject<RequestResponseAmqpLink>(
-                    timeout => ConnectionScope.OpenManagementLinkAsync(timeout, CancellationToken.None),
+                    linkTimeout => ConnectionScope.OpenManagementLinkAsync(operationTimeout, linkTimeout, CancellationToken.None),
```

Comment:
I guess the operationTimeout needs to be threaded through here because it is set at the producer/consumer level but someone might be sharing a connection?

Reply:
Exactly.   We need a timeout for the connection-level activities like the metadata queries.  We don't have a retry policy available until they're called - but we need a timeout for the connection that they share.